File: yh_tutorial/yh_tutorial_5/scripts/yh_server_5.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import rospy
from yh_tutorial_5.srv import yh_srv_5, yh_srv_5Response

def srvCallback(req):
    return yh_srv_5Response(req.a * req.b)

def yh_server():
    rospy.init_node("yh_server_5")
    service_server = rospy.Service("yh_service_5", yh_srv_5, srvCallback)
    rospy.spin()

if __name__ == "__main__":
    yh_server()

File: yh_tutorial/yh_tutorial_5/src/yh_server_5.cpp
#include "ros/ros.h"
#include "yh_tutorial_5/yh_srv_5.h"

bool srvCallback(yh_tutorial_5::yh_srv_5::Request &req, yh_tutorial_5::yh_srv_5::Response &res)
{
    res.result = req.a * req.b;

    return true;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "yh_server_5");
    ros::NodeHandle nh;

    ros::ServiceServer service_server = nh.advertiseService("yh_service_5", srvCallback);

    ROS_INFO("Service server ready.");

    ros::spin();

    return 0;
}


